### DISCUSSION TITLE
Async Reading Group on Consensus Protocols

### DISCUSSION THREAD

**Discussion by sa8** [consensus-shipyard/consensuslab#104] [Reading group]
We will use the discussion section in this repo to provide short summaries of consensus protocols that we read.
We will create a new discussion for each paper read and tag the "Journal Club" category.
We read these papers in the light of applying them to Filecoin's hierarchical consensus root protocol, i.e., top level. Informally the requirements that we look for when reading these papers are, by order of importance (in addition to the usual security properties considered by any consensus protocols):
- scaling to ten thousands nodes.
- censorship resistance. This specifically applies to WindowPosts that should be included in the chain 30 minutes after a sector is challenged on-chain.
- relatively low finality. 
  - Ideally, checkpoints coming from subnets should be included in the chain within 5 minutes (this may however be too strong of a requirement).
  - It should not be feasible to rewrite the history of a length > `PreCommitChallengeDelay` (currently 150 blocks).
- fairness.
- simplicity.

# List of papers

## Priority

- [ ] [Avalanche ](https://knowen-production.s3.amazonaws.com/uploads/attachment/file/1922/Snowflake%2Bto%2BAvalanche%2B-%2BA%2BNovel%2BMetastable%2BConsensus%2BProtocol%2BFamily.pdf)-> @xuechao2 reading
- [ ] [Spacemesh](https://drive.google.com/file/d/18I9GPebWqgpvusI1kMnAB9nayBbL-1qN/view) -> @sa8 reading [Summary](https://github.com/protocol/ConsensusLab/discussions/110)

## Already read (summary to appear)
- [ ] [Algorand](https://dl.acm.org/doi/pdf/10.1145/3132747.3132757): [Summary](https://github.com/protocol/ConsensusLab/discussions/114#discussion-4168594)
- [ ] [Colordag](https://eprint.iacr.org/2022/308.pdf): [Summary](https://github.com/protocol/ConsensusLab/discussions/105#discussion-4159532)
- [ ] [Order-Fair Consensus in the Permissionless Setting](https://eprint.iacr.org/2021/139.pdf): [Summary](https://github.com/protocol/ConsensusLab/discussions/109#discussion-4163204)
- [ ] [Ebb-and-Flow](https://arxiv.org/abs/2009.04987) and its follow-ups: [Summary](https://github.com/protocol/ConsensusLab/discussions/107#discussion-4160406)
    - [ ] [Checkpointed longest-chain](https://arxiv.org/abs/2010.13711) 
    - [ ] [Accountability gadgets](https://arxiv.org/abs/2105.06075)
- [ ] Parallel chain protocols: [Summary](https://github.com/protocol/ConsensusLab/discussions/106#discussion-4159980)
    - [ ] [Prism](https://arxiv.org/pdf/1810.08092.pdf)
    - [ ] [Ledger combiners](https://eprint.iacr.org/2020/675.pdf)
- [ ] [Snow family of protocols](https://arxiv.org/pdf/1906.08936.pdf): [Summary](https://github.com/protocol/ConsensusLab/discussions/111#discussion-4165874)
    - [ ] Snowball
    - [ ] Avalanche
    - [ ] Snowman (no full description available, only an [video](https://www.youtube.com/watch?v=uiNK9-W_2Nw)) 
- [ ] Longest-chain protocols: [Summary](https://github.com/protocol/ConsensusLab/discussions/112)
    - [ ] [PoSAT](https://arxiv.org/pdf/2010.08154.pdf)
    - [ ] [Ouroboros Family](https://aft.acm.org/wp-content/uploads/2019/10/Ouroboros_AFT19_Tutorial.pdf)
## De-prioritized papers
- [ ] [Hot-stuff](https://dl.acm.org/doi/pdf/10.1145/3293611.3331591) -> not scalable enough
- [ ] ETH 2.0 -> many attacks vector (cf papers by Stanford)
  - [ ] [Goldfish](https://arxiv.org/pdf/2209.03255.pdf): [Summary](https://github.com/protocol/ConsensusLab/discussions/178#discussion-4388642)
- [ ] [DAG-based protocols](https://dahliamalkhi.github.io/posts/2022/06/dag-bft/) ->not scalable enough
  - [ ] [Aleph](https://cardinal-cryptography.github.io/AlephBFT/)
  - [ ] [Multi-resource consensus](https://www.youtube.com/watch?v=GhpVDfhHvX0)?

**Comment by vukolic** [comment]
"Too long finality" is not in my opinion a sufficient, or perhaps detailed enough reason to dismiss longest chain protocols easily. How do we define exactly "too long finality"?

**Reply by sa8** [reply]
I've put it back in the list and I'll create another discussion for longest chain protocols where I'll add some numbers and we can discuss the different trade-offs from there!

**Comment by Amanuttam1192** [comment]
Maybe it can be be fully sufficient because the longest chain protocols may also have  a time enimity that can also be incurred,maybe i am wrong,if please correct me

**Reply by sa8** [reply]
Hi @Amanuttam1192, I'm not sure I understand what you mean by "time enimity." Would you mind clarifying?

**Reply by Amanuttam1192** [reply]
I am sorry for the typo,by enimity i meant to write entity

**Reply by sa8** [reply]
No worries :). I agree that there is some sort of time entity with longest chain protocols, however there is still the issue of [private attacks](https://dl.acm.org/doi/pdf/10.1145/3479722.3480996) regardless. In the case of Filecoin this can be more problematic than in proof-of-stake due to some storage constraints.

**Reply by Amanuttam1192** [reply]
well i think that will solve my query

**Comment by sa8** [comment]
other suggestion from Akosh: [Near protocol](https://near.org/blog/doomslug-comparison/)

**Reply by Amanuttam1192** [reply]
thanks a lot mam

**Comment by sa8** [comment]
Other suggestion: [Internet Computer](https://dfinity.org/pdf-viewer/pdfs/viewer?file=../library/dfinity-consensus.pdf). Although I think there was some criticism about their protocol not being fully-open (to verify).

**Reply by Amanuttam1192** [reply]
ok ok

**Comment by sa8** [comment]
Suggestion from @guy-goren https://www.ndss-symposium.org/ndss-paper/bobtail-improved-blockchain-security-with-low-variance-mining/
Could be adapted for PoS and maybe tipsets.
